integration/utils/follower.go

```diff
@@ -40,8 +41,13 @@ func WithInteval(interval time.Duration) followerOption {
 	return func(f *txFollowerImpl) { f.interval = interval }
 }
 
+func WithMetrics(m *metrics.LoaderCollector) followerOption {
+	return func(f *txFollowerImpl) { f.metrics = m }
+}
+
 type txFollowerImpl struct {
```

Comment:
The main reason for using mutex for height and block is because they are updated as a pair.  But I get your point, let me rewrite it in a slower but robust way -- we do not need that code to be performant but we need it to be robust and easy to understand.